interfaces/builtin/kernel_firmware_control.go

```diff
@@ -0,0 +1,54 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2024 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+ const kernelFirmwareControlSummary = `allows adding new path for firmware location`
+ 
+ const kernelFirmwareControlBaseDeclarationPlugs = `
+  kernel-firmware-control:
+    allow-installation: false
+    deny-auto-connection: true
+`
+
+const kernelFirmwareControlBaseDeclarationSlots = `
+  kernel-firmware-control:
+    allow-installation:
+      slot-snap-type:
+        - core
+    deny-auto-connection: true
+`
+
+const kernelFirmwareControlConnectedPlugAppArmor = `
+# Description: Allow adding new path for firmware location.
+/sys/module/firmware_class/parameters/path rw,
+`
+
+
+func init() {
+	registerIface(&commonInterface{
+		name:                  "kernel-firmware-control",
```

Comment:
maybe `kernel-firmware-location-control`? Otherwise the name would suggest that there is some level of control over kernel firmware by eg. sysfs helper, while there isn't any actually.

Reply:
the `kernel-firmware-control` was suggested by Alex and Samuele.

Reply:
Please don't take my suggested name as gospel ;) - @pedronis would be the ultimate authority on that - however I tend to agree with @bboozzoo that interface names should be as accurate as possible - how about `kernel-firmware-path-control`? Or if we want to call this `kernel-firmware-control` should it then be expanded to include direct firmware loading via https://www.kernel.org/doc/html/v4.14/driver-api/firmware/fallback-mechanisms.html#firmware-sysfs-loading-facility

Reply:
@alexmurray we don't want to create too many interfaces without reason, so my thinking is to start with just the path but expand this later?

Reply:
Sure, let's keep this current interface/PR to just what we know is required at this time.